saleor/graphql/attribute/types.py

```diff
@@ -656,3 +655,791 @@ class AttributeValueInput(BaseInputObjectType):
+    value = graphene.Field(
+        "saleor.graphql.product.types.ProductVariant",
+        description="The assigned product variant reference.",
+        required=False,
+    )
+
+    class Meta:
+        interfaces = [AssignedAttribute]
+        description = (
+            "Represents single product variant reference attribute." + ADDED_IN_322
+        )
+        doc_category = DOC_CATEGORY_ATTRIBUTES
+
+    @staticmethod
+    def resolve_value(
+        root: AssignedAttributeData, info: ResolveInfo
+    ) -> Promise[ChannelContext[product_models.ProductVariant]] | None:
+        if not root.values:
+            return None
+
+        channel_slug = root.attribute.channel_slug
+        attr_value = root.values[0].node
+
+        return (
+            ProductVariantByIdLoader(info.context)
+            .load(attr_value.reference_variant_id)
+            .then(
+                lambda product_variant: ChannelContext(
+                    node=product_variant, channel_slug=channel_slug
+                )
+            )
+        )
+
+
+class AssignedSingleCategoryReferenceAttribute(BaseObjectType):
+    value = graphene.Field(
+        "saleor.graphql.product.types.Category",
+        description="The assigned category reference.",
+        required=False,
+    )
+
+    class Meta:
+        interfaces = [AssignedAttribute]
+        description = "Represents single category reference attribute." + ADDED_IN_322
+        doc_category = DOC_CATEGORY_ATTRIBUTES
+
+    @staticmethod
+    def resolve_value(
+        root: AssignedAttributeData, info: ResolveInfo
+    ) -> Promise[product_models.Category] | None:
+        if not root.values:
+            return None
+        attr_value = root.values[0].node
+        return CategoryByIdLoader(info.context).load(attr_value.reference_category_id)
+
+
+class AssignedSingleCollectionReferenceAttribute(BaseObjectType):
+    value = graphene.Field(
+        "saleor.graphql.product.types.Collection",
+        description="The assigned collection reference.",
+        required=False,
+    )
+
+    class Meta:
+        interfaces = [AssignedAttribute]
+        description = "Represents single collection reference attribute." + ADDED_IN_322
+        doc_category = DOC_CATEGORY_ATTRIBUTES
+
+    @staticmethod
+    def resolve_value(
+        root: AssignedAttributeData, info: ResolveInfo
+    ) -> Promise[ChannelContext[product_models.Collection]] | None:
+        if not root.values:
+            return None
+
+        channel_slug = root.attribute.channel_slug
+        attr_value = root.values[0].node
+
+        return (
+            CollectionByIdLoader(info.context)
+            .load(attr_value.reference_collection_id)
+            .then(
+                lambda collection: ChannelContext(
+                    node=collection, channel_slug=channel_slug
+                )
+            )
+        )
+
+
+class AssignedMultiPageReferenceAttribute(BaseObjectType):
+    value = NonNullList(
```

Comment:
In this field, Saleor returns multiple values. 
```suggestion
    values = NonNullList(
```

Reply:
This is called `value` by design. Like we have the attribute, where the `value` is the list